common/processes/processes.cpp

```diff
@@ -45,6 +50,26 @@ error::Error MakeError(ProcessesErrorCode code, const string &msg) {
 	return error::Error(error_condition(code, ProcessesErrorCategory), msg);
 };
 
+void OutputHandler::operator()(const char *data, size_t size) {
+	if (size == 0) {
+		return;
+	}
+	// Get rid of exactly one trailing newline, if there is one. This is because
+	// we unconditionally print one at the end of every log line. If the string
+	// does not contain a trailing newline, add a "{...}" instead, since we
+	// cannot avoid breaking the line apart then.
+	string content(data, size);
+	if (content.back() == '\n') {
+		content.pop_back();
+	} else {
+		content.append("{...}");
+	}
+	auto lines = mender::common::SplitString(content, "\n");
+	for (auto line : lines) {
+		log::Info(prefix + line);
+	}
```

Comment:
Leaving this one for later :)

Reply:
I thought quite a bit about this when I first implemented it, and I think it's impossible to do perfectly, if we take into account that we need to consider limited memory. We are going to get data in blocks, whether we want to or not, and if there is an ultra long line, we either need to print it with `{...}`, or save it to memory, and print it once we get the `\n`.

Printing it partially is not possible, because log lines always insert newline, and even if they didn't, other log lines could be printed in between. And partial printing would not work for the JSON based deployment log either.

At least I couldn't think of anything besides those options, so that's what I concluded with.